src/server/server.go

```diff
@@ -59,10 +65,64 @@ func PprofMem(w http.ResponseWriter, r *http.Request) {
         runtime.GC()
 	w.Header().Add("Content-Type", "application/octet-stream")
         if err := pprof.WriteHeapProfile(w); err != nil {
-            log.Fatal("could not write memory profile: ", err)
+	    	log.Fatal("could not write memory profile: ", err)
         }
 }
 
+// Starts CPU profiling
+func PprofCpuStart(w http.ResponseWriter, r *http.Request) {
```

Comment:
The way I would probably do this:
1. have a separate function that starts profiling OR returns an error
2. this function will call the helper.   If there is an error, write back whatever it is with a status code.  Then you can handle all the errors in one place.